crates/jit-icache-coherence/src/lib.rs

```diff
@@ -0,0 +1,106 @@
+//! This crate provides utilities for instruction cache maintenance for JIT authors.
+//!
+//! In self modifying codes such as when writing a JIT, special care must be taken when marking the
+//! code as ready for execution. On fully coherent architectures (X86, S390X) the data cache (D-Cache)
```

Comment:
Not strictly necessary for this PR, but I wonder how RISC-V fits into this -- it looks like at the ISA level it has a `fence.i` instruction, so it is closer to AArch64 in this regard (weaker coherence by default). Is it enough to do the same `membarrier` calls as on `aarch64`? (cc @yuyang-ok)

In the absence of any other information, perhaps we could perform the same `membarrier` calls on RISC-V as we do on aarch64?

Reply:
I agree that we do need to do something, from what I've read RISCV is allowed to have incoherent I and D caches. From this [documentation](https://docs.kernel.org/riscv/features.html) of the kernel, it looks like CORE_SYNC is not yet implemented for RISCV.

I've tried to read the kernel a bit and I've found that they support a

Reply:
> Or to put it another way - an architecture having incoherent data and instruction caches does not imply that it behaves in exactly the same way as the 64-bit Arm architecture (and hence requiring exactly the same sequence of actions); possibly there are nuances.

Yeah, that's right, we should go and double check that!

I've opened https://github.com/bytecodealliance/wasmtime/issues/5033 to track this, but I'm going to look at the ISA manual to check if they guarantee anything like that.